syntax = "proto3";
package weather;

message WeatherRequest {
  string city_name = 1;
}

message WeatherResponse {
  string city_name = 1;
  double temperature = 2;
  string weather_description = 3;
  double humidity = 4;
  double wind_speed = 5;
  string timestamp = 6;
}

message WeatherHistoryRequest {
  string city_name = 1;
  string start_time = 2;
  string end_time = 3;
}

message WeatherHistoryResponse {
  repeated WeatherResponse readings = 1;
}

service WeatherService {
  rpc GetWeather (WeatherRequest) returns (WeatherResponse);
  rpc GetWeatherHistory (WeatherHistoryRequest) returns (WeatherHistoryResponse);
}